src/Datadog.Trace/Logging/DatadogLogging.cs

```diff
@@ -170,5 +143,49 @@ public static ILogger For<T>()
         {
             return GetLogger(typeof(T));
         }
+
+        private static string GetLogDirectory()
+        {
+            var nativeLogFile = Environment.GetEnvironmentVariable(ConfigurationKeys.ProfilerLogPath);
+            string logDirectory = null;
+
+            if (!string.IsNullOrEmpty(nativeLogFile))
+            {
+                logDirectory = Path.GetDirectoryName(nativeLogFile);
+            }
+
+            // This entire block may throw a SecurityException if not granted the System.Security.Permissions.FileIOPermission
+            // because of the following API calls
+            //   - Directory.Exists
+            //   - Environment.GetFolderPath
+            //   - Path.GetTempPath
+            if (logDirectory == null)
+            {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    var windowsDefaultDirectory = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData), @"Datadog .NET Tracer", "logs");
+                    if (Directory.Exists(windowsDefaultDirectory))
+                    {
+                        logDirectory = windowsDefaultDirectory;
+                    }
+                }
+                else
+                {
+                    // either Linux or OS X
+                    if (Directory.Exists(NixDefaultDirectory))
+                    {
+                        logDirectory = NixDefaultDirectory;
+                    }
+                }
+            }
+
+            if (logDirectory == null)
+            {
+                // Last effort at writing logs
+                logDirectory = Path.GetTempPath();
```

Comment:
Let's create a sub-directory under the temp path so our log files don't get lost among hundreds of files. Something like `datadog-dotnet-tracer`.

```suggestion
                logDirectory = Path.Combine(Path.GetTempPath(), "datadog-dotnet-tracer");
```

Reply:
In the event there is no existing path, ok.